### PR TITLE
#15 json naming convention

### PR DESCRIPTION
**Pull request by evolverine**
I'm suggesting a naming convention, where all the json files are named 'deck.json', which can simplify naming issues especially for the importer. Then it's both the folder they're in and also the github repo that offer the user the knowledge of which deck they have there.

### CODE REVIEW

**Comment by Stvad** [comment]
This, makes sense, but I'd like to ensure backward compatibility on import. I.e. the import with the old convention should still work.

File: crowd_anki/export/anki_exporter.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def export_to_directory(self, deck: AnkiDeck, output_dir=Path("."), copy_media=T
         deck = deck_initializer.from_collection(self.collection, deck.name)
         self.last_exported_count = deck.get_note_count()
 
-        deck_filename = deck_directory.joinpath(deck_fsname).with_suffix(DECK_FILE_EXTENSION)
+        deck_filename = deck_directory.joinpath('deck').with_suffix(DECK_FILE_EXTENSION)
```
Comment (Stvad) [inline crowd_anki/export/anki_exporter.py]: Please extract the actual name used to a constant in https://github.com/Stvad/CrowdAnki/blob/master/crowd_anki/utils/constants.py

File: crowd_anki/importer/anki_importer.py [hunk 2]
```diff
@@ -42,7 +42,7 @@ def load_from_directory(self, directory_path, import_media=True):
             aqt.mw.progress.start(immediate=True)
 
         try:
-            self.load_from_file(directory_path.joinpath(directory_path.name).with_suffix(DECK_FILE_EXTENSION))
+            self.load_from_file(directory_path.joinpath('deck').with_suffix(DECK_FILE_EXTENSION))
```
Comment (Stvad) [inline crowd_anki/importer/anki_importer.py]: You should update the docstring of the function to describe new convention

**Comment by Stvad** [comment]
Btw, I see several merge commits on the branch - please look into using the `git pull --rebase` instead to maintain the git history cleaner

**Comment by Stvad** [comment]
Also, the Readme should be updated to reflect new convention

**Comment by evolverine** [comment]
Apologies for the delay. I will look at implementing your suggestions next weekend.

**Comment by evolverine** [comment]
> 
> 
> This, makes sense, but I'd like to ensure backward compatibility on import. I.e. the import with the old convention should still work.

I've implemented this in the latest commit (and tested it too, locally; seems to work just fine).

File: crowd_anki/export/anki_exporter.py (hunk 1 above)
Comment (evolverine) [reply crowd_anki/export/anki_exporter.py]: good point, done in the latest commit.

File: crowd_anki/importer/anki_importer.py (hunk 2 above)
Comment (evolverine) [reply crowd_anki/importer/anki_importer.py]: done in the latest commit.

**Comment by evolverine** [comment]
> 
> 
> Also, the Readme should be updated to reflect new convention

I've checked the Readme, and the only places that the file naming convention comes up seems to be in the screenshots. And I haven't changed them, partly because I didn't want my git username and projects to replace yours; it only makes sense that they reflect your workspace, given that you are the author of the plugin, at least that's what I reasoned. But let me know if you'd like these updated too, and I can do that.

**Comment by Stvad** [comment]
Hey, sorry for the delay in looking at this. I've checked it and merged it with minor updates

**Comment by evolverine** [comment]
Great, thank you!

**Pull request closed without merging**